### ISSUE TITLE
"Custom properties fallback values" support

### ISSUE THREAD

**Issue by emmanuel-ma** [jhildenbiddle/css-vars-ponyfill#37]
Hi guys,

I have a user case where I need to use "Custom properties fallback values" on this way:

```
background-color: var(--my-var, var(--my-background)); /* --my-background if my-var is not defined */

```

On modern browsers is working fine but on IE11 does not work using this library. 

Does someone know if the css-vars-ponyfill supports this feature? If doesn't, do you know some workaround?

**Comment by jhildenbiddle** [comment]
Hi @emmanuel-ma.

The ponyfill has supported this feature since 1.0. I've confirmed it is working as expected in the linked demo:

* https://codepen.io/jhildenbiddle/pen/ZxYJrR/

Notice the fallback test on line 17 in the CSS panel:

```css
  font-family: var(--fail, var(--font-family));
```

Notice how this declaration is transformed to `font-family: sans-serif;` in the output.

**Comment by emmanuel-ma** [comment]
Hi @jhildenbiddle ,

Thanks for your support. You are right, the fallback values work great, your library is a beauty.

I am working with a React project and the issue was related with the CSS resource link. If its *href* property is linked to a non-existent CSS file, the HTTP request status is 200 and the retrieved content is the content of the index.html file of the project. I expected a 404 status and an `options.onError` event but I don't find out the origin of this behavior yet. In the meantime, I had to use the following code to solve the issue.
```
cssVars({
      // Exclude Bootstrap & Google CSS resources
      exclude: '[href*=bootstrap],[href*=google]',
      onSuccess(cssText, node, url) {
        // If the HTTP request status to a non-existent CSS file is 200,
        // then we need to verify that the content of the file is not
        // the 'public/index.html' content.
        if (cssText && cssText.startsWith('<!DOCTYPE html>')) {
          // Excluding link tag
          return ':root {}'; // empty CSS variables
        }
        return cssText;
      }
    });
```
Have you had a similar case?

**Comment by jhildenbiddle** [comment]
Interen

**Comment by jhildenbiddle** [comment]
I haven't run into that, but it makes sense. Sounds like your app is just configured to redirect all 404 errors to `index.html`. 

I think your solution works well, although I'd propose a few small tweaks:

1. Trim `cssText` and simplify the test to just an open bracket. This will account for spaces before the tag, upper/lower case, and other doctypes.
1. Replace [`startWith()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith) (which isn't supported in IE or Safari < 9) for something more legacy-friendly.

```javascript
cssVars({
  exclude: '[href*=bootstrap],[href*=google]',
  onSuccess(cssText, node, url) {
    const isHTML = cssText.trim().charAt(0) === '<';
    return isHTML ? false : cssText;
  }
});
```

**Comment by brodybits** [comment]
Shouldn't this be documented?

**Comment by jhildenbiddle** [comment]
Better yet, perhaps the above check (first character of the `trim()`ed XHR response !== "<") should just be baked into the ponyfill. It seems like a reasonable check, since I can't imagine a scenario a CSS file would being with an open bracket.

Thoughts?

**Comment by brodybits** [comment]
Sounds good to me. I neglected to mention that I am actually a newbie with CSS vars, hope we get feedback from people with more expertise. Also hoping to see a more appropriate label on this issue:)

**Comment by jhildenbiddle** [comment]
@emmanuel-ma --

I've updated the ponyfill to perform the above test automatically. You should be able to remove your `onSuccess` callback  after you've updated the ponyfill to 1.14.0.

FWIW, the update was actually made in the [get-css-data](https://github.com/jhildenbiddle/get-css-data) lib which handles fetching `<link>` CSS text for the ponyfill.

Hope this addresses your issue. Let me know, then I'll close this issue.

Thanks!

**Comment by emmanuel-ma** [comment]
Hey @jhildenbiddle ,

With the new version 1.14.0 all is working great! Thanks, you really rock!

**Comment by jhildenbiddle** [comment]
Excellent! Good to hear the new version is working for you, @emmanuel-ma.

You were right about the `options.onSuccess` callback not handling a return value of `false` as it should have. This has been fixed in 1.15.0, and the callback now handles other falsey values properly as well (e.g. `null`, `0`, `""`).

Thanks for taking the time to file these issues. They make the ponyfill better for everyone! 😄
